primitives/sr-io/src/lib.rs

```diff
@@ -770,8 +771,10 @@ pub fn panic(info: &core::panic::PanicInfo) -> ! {
 	}
 }
 
+/// A default OOM handler for WASM environment.
 #[cfg(all(not(feature = "disable_oom"), not(feature = "std")))]
 #[alloc_error_handler]
+#[allow(improper_ctypes)] // core::alloc::Layout is not FFI-safe.
```

Comment:
Hm? 

1. I thought that FFI safeness has to do with `extern "C"`, since `extern` should assume Rust ABI.
2. We don't need `extern` here in the first place, I think

Reply:
I have no idea, just repeating what the warning says :)

Reply:
Given this: https://github.com/rust-lang/rust/issues/51540

The signature should be changed to `fn oom(_: core::alloc::Layout) -> !`